model/testing/src/icon4py/model/testing/pytest_config.py

```diff
@@ -123,6 +123,9 @@ def pytest_addoption(parser):
 
 
 def pytest_runtest_setup(item):
+    for marker in item.own_markers:
+        if "embedded" in (marker.name, item.config.getoption("--backend")):
```

Comment:
I still do not understand this condition. I suspect you mean:
```suggestion
        if marker.name.startwith("embedded") and item.config.getoption("--backend") == "embedded":
```